app/src/main/java/com/google/samples/apps/sunflower/adapters/PlantAdapter.kt

```diff
@@ -47,29 +44,23 @@ class PlantAdapter : ListAdapter<Plant, PlantAdapter.ViewHolder>(PlantDiffCallba
     }
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-        getItem(position).apply {
-            holder.itemView.tag = this
-            with(holder.binding) {
-                setVariable(BR.vm, ItemViewModel(this@apply, onClickListener))
-                executePendingBindings()
-            }
+        val plant = getItem(position)
+        holder.apply {
+            bind(plant)
+            itemView.tag = plant
+            itemView.setOnClickListener(onClickListener)
```

Comment:
with `bind()` is ok, however, I can't understand why keep `listener` out of binding instead in xml?

Reply:
Reverted the change, so now the listener is passed as a binding variable.